### PR TITLE
Fix subworkflows based on json schema

### PR DESCRIPTION
**Pull request by mashehu**
Yet another json schema, this time for subworkflows (needed so they are all nicely rendered on the website).
In order for the two schemas to behave nicely, I had to split them into two files and put them into the respective subfolders. It is very similar to the schema for modules, biggest difference is that I don't require types for the input and output, because it felt like 90% was some kind of `map` (you can still give them and they will be rendered on the homepage).

I fixed the broken meta.yaml according to  this new rules.

I also added the hint for the yaml-language server, to nicely show the problems already in the editor (thanks @Emiller88 for pointing this out)

### CODE REVIEW

File: subworkflows/yaml-schema.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                                "description": "Type of the input channel"
+                            },
+                            "description": {
+                                "type": "string",
+                                "description": "Description of the input channel"
+                            },
+                            "pattern": {
+                                "type": "string",
+                                "description": "Pattern of the input channel, given in Java glob syntax"
+                            }
```
Comment (mirpedrol) [inline subworkflows/yaml-schema.json:57]: We have also a "default" field in some of them, do we want to check this one too?

**Review by mirpedrol — APPROVED** [review]
LGTM :)

File: subworkflows/yaml-schema.json (hunk 1 above)
Comment (mashehu) [reply subworkflows/yaml-schema.json:57]: Good to know, I'll add it back then.

**Comment by maxulysse** [comment]
cf https://github.com/nf-core/modules/pull/3340

**Pull request merged**